include/MainWindow.h

```diff
@@ -57,7 +59,7 @@ class MainWindow : public QMainWindow
 public:
 	QMdiArea* workspace()
 	{
-		return m_workspace;
+		return reinterpret_cast<QMdiArea*>(m_workspace);
```

Comment:
Seems like this doesn't work:

`invalid ‘static_cast’ from type ‘lmms::gui::CustomQMdiArea*’ to type ‘QMdiArea*’`

Reply:
Hm, try `dynamic_cast` instead. You usually wouldn't use `reinterpret_cast` in this case.

Reply:
This causes an error also:

`return dynamic_cast<QMdiArea*>(m_workspace);`

Reply:
Ah I see what is going on. You need to make two changes: one is to use `static_cast`, and the other is to declare `CustomQMdiArea` before `MainWindow`, ideally in a separate header file and then include it into `MainWindow.h`. I also think `MovableQMdiArea` is a better name, what do you think?

Reply:
Is making new files for this necessary? I would rather put `QMdiArea* workspace()`'s definition inside `MainWindow.cpp`, that way `static_cast` will work, but If you think it's necessary to put that in a new file, then sure! I agree on the name change, "Movable" gives a better description than "Custom".

Reply:
I think its personal preference really, but I don't think we have any conventions that address this in particular. Also consider if you want to move the custom class to be declared inside `MainWindow`. That way, you don't have to put it in a new file, everything will seem more "together" rather than having distinct classes, and the order in which you define both `MainWindow` and the `QMdiArea` wont matter at all.

Reply:
Okay, I will move the `MovableQMdiArea` class in to the `MainWindow` class